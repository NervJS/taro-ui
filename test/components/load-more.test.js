import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtLoadMore from '../../.temp/components/load-more/index'

describe('AtLoadMore Snap', () => {
  it('render initial AtLoadMore', () => {
    const component = renderToString(<AtLoadMore />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props customStyle', () => {
    const component = renderToString(<AtLoadMore customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props className', () => {
    const component = renderToString(<AtLoadMore className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props status === more', () => {
    const component = renderToString(<AtLoadMore status='more' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props status === loading', () => {
    const component = renderToString(<AtLoadMore status='loading' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props status === noMore', () => {
    const component = renderToString(<AtLoadMore status='noMore' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props moreText ', () => {
    const component = renderToString(
      <AtLoadMore moreText='moreText' status='more' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props loadingText ', () => {
    const component = renderToString(
      <AtLoadMore loadingText='loadingText' status='loading' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props noMoreText ', () => {
    const component = renderToString(
      <AtLoadMore noMoreText='noMoreText' status='noMore' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props noMoreTextStyle', () => {
    const component = renderToString(<AtLoadMore noMoreTextStyle='color:red' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtLoadMore -- props moreBtnStyle', () => {
    const component = renderToString(<AtLoadMore moreBtnStyle='color:red' />)
    expect(component).toMatchSnapshot()
  })
})
