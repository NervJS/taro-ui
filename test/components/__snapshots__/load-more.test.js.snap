// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtLoadMore Snap render AtLoadMore -- props className 1`] = `"<div class=\\"at-load-more test\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props loadingText  1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-activity-indicator at-activity-indicator--center at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><span class=\\"taro-text at-activity-indicator__content\\">loadingText</span></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props moreBtnStyle 1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div style=\\"color:red\\" class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props moreText  1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">moreText</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props noMoreText  1`] = `"<div class=\\"at-load-more\\"><span class=\\"taro-text at-load-more__tip\\">noMoreText</span></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props noMoreTextStyle 1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props status === loading 1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-activity-indicator at-activity-indicator--center at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><span class=\\"taro-text at-activity-indicator__content\\">加载中</span></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props status === more 1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;

exports[`AtLoadMore Snap render AtLoadMore -- props status === noMore 1`] = `"<div class=\\"at-load-more\\"><span class=\\"taro-text at-load-more__tip\\">没有更多</span></div>"`;

exports[`AtLoadMore Snap render initial AtLoadMore 1`] = `"<div class=\\"at-load-more\\"><div class=\\"at-load-more__cnt\\"><div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">查看更多</div></div></div></div>"`;
